src/py/mat3ra/made/tools/build/utils.py

```diff
@@ -0,0 +1,87 @@
+from typing import List, Dict
+
+import numpy as np
+from mat3ra.made.material import Material
+from mat3ra.made.utils import filter_array_with_id_value_by_ids
+
+from mat3ra.made.tools.utils import convert_basis_to_crystal
+
+
+def resolve_close_coordinates(basis: Dict, distance_tolerance: float = 0.01) -> Dict:
+    """
+    Resolve close coordinates in the basis by replacing them with the latest one.
```

Comment:
Find all coordinates that are within distance tolerance and only keep the last one